src/agent/core.jl

```diff
@@ -20,6 +20,7 @@ export @agent,
 using UUIDs
 
 using Dates: Dates
+import Base.getindex
```

Comment:
Similar to what we discussed on importing `handle_message` => no need to import and explicitely make a method for `Base.getindex` for clarity?